registry/storage/registry.go

```diff
@@ -46,6 +52,23 @@ func DisableDigestResumption(registry *registry) error {
 	return nil
 }
 
+// DisableManifestURLs is a functional option for NewRegistry. It causes the
+// registry to reject pushed manifests containing URLs.
+func DisableManifestURLs(registry *registry) error {
```

Comment:
O, these are separate features? Why not just have a `deny` section?

```
validation:
  enabled: true # enabled turns on the feature
  manifests:
    urls:
      deny:
        - *
```

Typically, we use `enabled`/`disabled` as a feature switch. Having mean that "urls are disabled" is confusing.